### ISSUE TITLE
Use SPDX license descriptors by default

### ISSUE THREAD

**Issue by dbaynard** [commercialhaskell/stack-templates#136]
At least the default template uses `BSD3` for a license identifier. Cabal now supports SPDX license identifiers (and stack is getting features which also support this, e.g. as arguments to `ls dependencies`). It may be good for the templates to use the SPDX equivalent (`BSD-3-Clause`) instead.

**Comment by dbaynard** [comment]
(I'm happy to PR if desired; it's a simple grep)

**Comment by BinderDavid** [comment]
+1

This also requires to update the cabal-versions in the templates to at least 2.2. Otherwise the following warning will be emitted:

```
Warning: 'license: BSD-3' is not a recognised license. The known licenses are:                                                        
GPL, GPL-2, GPL-3, LGPL, LGPL-2.1, LGPL-3, AGPL, AGPL-3, BSD2, BSD3, MIT, ISC,                                                        
MPL-2.0, Apache, Apache-2.0, PublicDomain, AllRightsReserved, OtherLicense
```

An update would remove a little bit of friction that one encounters when trying to update the cabal version to 2.2. in order to use e.g. cabal common stanzas. In that case cabal complains about non SPDX compliant Licence specifiers.

**Comment by mpilgrem** [comment]
If the `BSD3` are replaced by `BSD-3-Clause` then Stack's in-built Hpack will specify `cabal-version: 2.2`. I think `Cabal-2.2.0.0` was released in March 2018 and included with GHC 8.4.1 (also released in March 2018). Only resolvers before `lts-12.1` are pre-GHC 8.4.1. I can't see any harm in bumping to `BSD-3-Clause` and, consequently, `cabal-version: 2.2`.

**Comment by mpilgrem** [comment]
With no objections and the passage of another year, I have put this through. Stack still supports users of GHC < 8.4.1, but the number of such people inconvenienced by this is likely small.
